### DISCUSSION TITLE
How do I restore a hidden commit (i.e. "un-abandon" it)? Why don't `branch set` and `edit` do this automatically?

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#4537] [Q&A] [answered]
[While writing this, I answered the first part of my own question, so I'll add that as an answer. The question about `branch set` and `edit` remains.]

I screwed up a rebase, and after a couple `jj op restore <id>`s, I eventually managed to find a repository state in which a particular commit I had lost was visible, but still hidden, using `jj log -r <commit-hash>`. I tried to "restore" the commit by assigning a branch to it, but that didn't work:

```
❯ jj log -r <branchname>
○  twtyyppm hidden kstrand@rigetti.com 2024-09-25 13:47:06 <branchname>* 5b942ee9
│  <description>
~
```

Even making that my working copy doesn't un-hide it:

```
❯ jj edit <branch>
Working copy now at: twtyyppm hidden 5b942ee9 <branch>* | <description>
Parent commit      : puxvrypl hidden b3d4fca2 <description>
Added 0 files, modified 5 files, removed 0 files
```

**Comment by BatmanAoD** [comment]
To answer the first part, how to restore the commit, making a `new` commit on top of the hidden one restored it:

```
❯ jj new
Working copy now at: kymqltqp b6ec8e3f (empty) (no description set)
Parent commit      : twtyyppm 5b942ee9 <branch>* | <description>
```

**Comment by yuja** [comment, marked as answer]
69edc7f2df298c94c27e2dc392c25e732bf5b5be should fix the problem of `jj edit`. `branch set`/`move` have the same problem. They should probably re-add visible heads or reject hidden commits.
